app/[id]+api.ts

```diff
@@ -1,79 +1,86 @@
-import { deleteGuest, getGuest, Guest } from '../database/guests';
+import { deleteGuestInsecure, getGuestInsecure } from '../database/guests';
+import { ExpoApiResponse } from '../ExpoApiResponse';
+import { Guest, guestsSchema } from '../migrations/00000-createTableGuests';
+
+type ExpoResponseBodyGet =
+  | {
+      guest: Guest;
+    }
+  | { error: string };
+
+export async function GET(
+  request: Request,
+  { id }: { id: string },
+): Promise<ExpoApiResponse<ExpoResponseBodyGet>> {
+  const guest = await getGuestInsecure(Number(id));
 
-export function GET(request: Request, { id }: { id: string }) {
-  const guest = getGuest(Number(id));
-  console.log(guest);
   if (!guest) {
-    return Response.json(
+    return ExpoApiResponse.json(
       { error: `No guest with id ${id} found` },
       { status: 404 },
     );
   }
-  return Response.json(guest);
+  return ExpoApiResponse.json({ guest: guest });
 }
 
-export function DELETE(request: Request, { id }: { id: string }) {
-  console.log('delete guest', id);
+type ExpoResponseBodyDelete =
+  | {
+      guest: Guest;
+    }
+  | { error: string };
 
-  const guests = deleteGuest(Number(id));
+export async function DELETE(
+  request: Request,
+  { id }: { id: string },
+): Promise<ExpoApiResponse<ExpoResponseBodyDelete>> {
+  const guest = await deleteGuestInsecure(Number(id));
+
+  if (!guest) {
+    return ExpoApiResponse.json(
+      { error: `Guest ${id} not found` },
+      { status: 404 },
+    );
+  }
 
-  return Response.json(guests);
+  return ExpoApiResponse.json({ guest: guest });
 }
```

Comment:
seems like the functions in this file are in the wrong order (GET, DELETE, PUT)

they should follow the same CRUD order that we have in our database functions files

(this is also inconsistent in the Next.js example: https://github.com/upleveled/next-js-example-fall-2024-atvie/blob/af2e2c1b65cd97de2af842fd109b294524acf256/app/api/animals/%5BanimalId%5D/route.ts - I will fix it there)

Reply:
Changed the order to follow CRUD
[cb39815](https://github.com/upleveled/expo-example-spring-2024-atvie/pull/4/commits/cb39815b331bbc8df2ef7e4ff95b26775aebfcdc)

Reply:
to be clear, it's not the CRUD order Create Read Update Delete, we flip Create and Read so that Read starts

Reply:
fixed in ba3efd5